src/LowMachEquationSystem.C

```diff
@@ -2051,20 +2059,28 @@ MomentumEquationSystem::register_wall_bc(
 
         // Assemble wall stresses via the edge algorithm.
         auto& solverAlgMap = solverAlgDriver_->solverAlgorithmMap_;
-        AssembleElemSolverAlgorithm* solverAlg = nullptr;
+        stk::topology elemTopo = get_elem_topo(realm_, *part);
+        AssembleFaceElemSolverAlgorithm* faceElemSolverAlg = nullptr;
         bool solverAlgWasBuilt = false;
 
-        std::tie(solverAlg, solverAlgWasBuilt) =
-          build_or_add_part_to_face_bc_solver_alg(
-            *this, *part, solverAlgMap, "wall_fcn");
+        std::tie(faceElemSolverAlg, solverAlgWasBuilt) =
+          build_or_add_part_to_face_elem_solver_alg(
+            wfAlgType, *this, *part, elemTopo, solverAlgMap, "wall_func");
 
-        ElemDataRequests& dataPreReqs = solverAlg->dataNeededByKernels_;
-        auto& activeKernels = solverAlg->activeKernels_;
+        if (userData.isNoSlip_) {
+          notProjectedPart_.push_back(part);
+          ablWallNodeMask_.reset(new MomentumABLWallFuncMaskUtil(realm_, part));
```

Comment:
Need to add the part to the existing mask utility if it exists. This will just operate on the last part in the no slip wall function is applied to.